gamemode/sv_player.lua

```diff
@@ -203,81 +203,243 @@ function GM:PlayerDeathSound()
+
+	if shuffled then
+		ShuffleTable(tbl)
+	end
+
+	return tbl
 end
 
+
+-- Generate points next to and above the spawn that we can test for suitability (a "3x3 grid")
+local function PointsAroundSpawn(spwn)
+	if !IsValid(spwn) then return {} end
+	local pos = spwn:GetPos()
+
+	local w, h = 50, 72 -- Increased from the default 36, 72 as it seems to work better with Prophunters.
+
+	-- all rigged positions
+	-- could be done without typing them out, but would take about as much time
+	return {
+		pos + Vector( w,  0,  0),
+		pos + Vector( 0,  w,  0),
+		pos + Vector( w,  w,  0),
+		pos + Vector(-w,  0,  0),
+		pos + Vector( 0, -w,  0),
+		pos + Vector(-w, -w,  0),
+		pos + Vector(-w,  w,  0),
+		pos + Vector( w, -w,  0)
+	};
+end
+
+
+function GM:PlayerSelectSpawn(ply)
+	local ply_team = ply:Team()
+	-- 3 = Props
+	-- 2 = Hunters
+	-- 1 = Spectators
+
+	-- Should be true when the first player joins the game
+	if !self.SpawnPoints then
+		self.SpawnPoints = {}
+	end
+
+	-- Should be true for each first player on a team
+	if !self.SpawnPoints[ply_team] || (table.IsEmpty(self.SpawnPoints[ply_team])) || (!IsTableOfEntitiesValid(self.SpawnPoints[ply_team])) then
+
+		self.SpawnPoints[ply_team] = GetSpawnEnts(ply_team, true, false)
+
+		-- One might think that we have to regenerate our spawnpoint
+		-- cache. Otherwise, any rigged spawn entities would not get reused, and
+		-- MORE new entities would be made instead. In reality, the map cleanup at
+		-- round start will remove our rigged spawns, and we'll have to create new
+		-- ones anyway.
+	end
+
+	if table.IsEmpty(self.SpawnPoints[ply_team]) then
+		Error("No spawn entity found!\n")
+		return
+	end
+
+	-- Just always shuffle, it's not that costly and should help spawn
+	-- randomness.
+	ShuffleTable(self.SpawnPoints[ply_team])
+
+	-- Optimistic attempt: assume there are sufficient spawns for all and one is
+	-- free
+	for _, spwn in pairs(self.SpawnPoints[ply_team]) do
+		if self:IsSpawnpointSuitable(ply, spwn, false) then
+			return spwn
+		end
+	end
+
+	-- That did not work, so now look around spawns
+	local picked = nil
+
+	for _, spwn in pairs(self.SpawnPoints[ply_team]) do
+		picked = spwn -- just to have something if all else fails
+
+		-- See if we can jury rig a spawn near this one
+		local rigged = PointsAroundSpawn(spwn)
+		for _, rig in pairs(rigged) do
+			if self:IsSpawnpointSuitable(ply, rig, false, true) then
+				local spawn_type
+				if ply:Team() == 3 then
+					spawn_type = "info_player_terrorist"
+				elseif ply:Team() == 2 then
+					spawn_type = "info_player_counterterrorist"
+				else
+					spawn_type = "info_player_start"
+				end
+				local rig_spwn = ents.Create(spawn_type)
+				if IsValid(rig_spwn) then
+					rig_spwn:SetPos(rig)
+					rig_spwn:Spawn()
+
+					ErrorNoHalt("PROPHUNTERS WARNING: Map has too few spawn points, using a rigged spawn for ".. tostring(ply:Nick()) .. "\n")
+
+					self.HaveRiggedSpawn = true
+					return rig_spwn
+				end
+			end
+		end
+	end
+
```

Comment:
At the very last final step where it's known that players will be killed, could you send a message to at least the admins to say "something's wrong"?

Reply:
Added a system for this.